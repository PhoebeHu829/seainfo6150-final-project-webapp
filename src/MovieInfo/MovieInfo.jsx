import React, { useState, useEffect } from 'react'
import DetailedInfo from '../DetailedInfo/DetailedInfo';
import Loading from '../Loading/Loading';

const MovieInfo = ({ movieId, addFav, fav, index_id_map, id_index_map }) => {
    const [movieInfo, setMovieInfo] = useState(null);

    useEffect(() => {
        setMovieInfo(null)
        fetch('https://api.themoviedb.org/3/movie/' + movieId + '?api_key=c1a32c68f363e400355e7be1a602cc66&language=en-US')
            .then(res => res.json())
            .then((data) => {
                //console.log(data);
                setMovieInfo(data);
            })
    }, [movieId])

    return (
        <>
            {movieInfo ?
                <DetailedInfo movie={movieInfo} addFav={addFav}
                    fav={fav} index_id_map={index_id_map}
                    id_index_map={id_index_map} />
                : <Loading />}
        </>
    )
}

export default MovieInfo
