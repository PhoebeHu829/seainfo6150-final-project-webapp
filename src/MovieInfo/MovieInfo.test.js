import React from 'react';
import MovieInfo from './MovieInfo.jsx';

describe('MovieInfo tests', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MovieInfo
        movieId={'399566'}
        addFav={() => {}}
        fav={[]}
        index_id_map={{ 1: '399566' }}
        id_index_map={{ 399566: '1' }}
        totalNum={1}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
