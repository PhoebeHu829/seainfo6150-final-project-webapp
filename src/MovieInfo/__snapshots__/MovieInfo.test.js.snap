// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo tests renders correctly 1`] = `
<div>
  <div
    class="loading"
  >
    <img
      alt="loading"
      src="https://i0.wp.com/www.cssscript.com/wp-content/uploads/2015/11/ispinner.jpg?fit=400%2C298&ssl=1"
    />
  </div>
</div>
`;
